shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java

```diff
@@ -33,22 +35,41 @@
 
   @NonNull private final FlutterJNI flutterJNI;
 
-  @NonNull private final ConcurrentHashMap<String, HandlerInfo> messageHandlers;
+  /**
+   * Maps a channel name to an object that contains the task queue and the handler associated with
+   * the channel.
+   *
+   * <p>Reads and writes to this map must lock {@code handlersLock}.
+   */
+  @NonNull
+  private final ConcurrentHashMap<String, HandlerInfo> messageHandlers = new ConcurrentHashMap<>();
+
+  /**
+   * Maps a channel name to queue of task dispatchers. This queue is processed when the channel
+   * handler is registered.
+   *
+   * <p>Reads and writes to this map must lock {@code handlersLock}.
+   */
+  @NonNull
+  private final ConcurrentHashMap<String, List<DelayedMessageInfo>> delayedTaskDispatcher =
```

Comment:
Update the name and comment now that this no longer holds task dispatchers.

This can also be a non-concurrent `HashMap`